packages/mobile/src/components/lineup-tile/VanityMetrics.tsx

```diff
@@ -0,0 +1,40 @@
+import type { ReactNode } from 'react'
+
+import type { ID } from '@audius/common/models'
+import { cacheUsersSelectors } from '@audius/common/store'
+import { TouchableOpacity } from 'react-native'
+import { useSelector } from 'react-redux'
+
+import type { IconComponent } from '@audius/harmony-native'
+import { Flex, Text } from '@audius/harmony-native'
+
+const { getUser } = cacheUsersSelectors
+
+type VanityMetricProps = {
+  icon?: IconComponent
+  children?: ReactNode
+  onPress?: () => void
+  disabled?: boolean
+}
+
+export const VanityMetric = (props: VanityMetricProps) => {
+  const { icon: Icon, children, onPress, disabled } = props
+
+  return (
+    <TouchableOpacity onPress={onPress} disabled={disabled}>
+      <Flex row alignItems='center' gap='xs'>
+        {Icon ? <Icon size='s' color='subdued' /> : null}
+        <Text size='xs' color='subdued'>
+          {children}
+        </Text>
+      </Flex>
+    </TouchableOpacity>
+  )
+}
+
+export const UserName = (props: { userId: ID }) => {
```

Comment:
This feels more generic. Does it belong somewhere else, or do we have something already? Nbd tho

Reply:
yeah i totally agree, this should be put somewhere generic. I was trying to use UserLink, but it doesnt quite work in this context